plugin/trino-ignite/src/test/java/io/trino/plugin/ignite/TestingIgniteServer.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.ignite;
+
+import io.trino.testng.services.ManageTestResources;
+
+import javax.annotation.concurrent.GuardedBy;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.Statement;
+
+import static io.trino.plugin.ignite.TestIgniteContainer.HTTP_PORT;
+import static java.lang.String.format;
+import static org.testcontainers.utility.MountableFile.forClasspathResource;
+
+@ManageTestResources.Suppress(because = "The actual server is kept on a static field and instance of this class is used for reference counting only")
+public class TestingIgniteServer
+{
+    @GuardedBy("this")
+    private static int referenceCount;
```

Comment:
Can you please the below class to `trino-testing-services` and use it here?
```
/*
 * Copyright Starburst Data, Inc. All rights reserved.
 *
 * THIS IS UNPUBLISHED PROPRIETARY SOURCE CODE OF STARBURST DATA.
 * The copyright notice above does not evidence any
 * actual or intended publication of such source code.
 *
 * Redistribution of this material is strictly prohibited.
 */

package com.starburstdata.presto.testing;

import javax.annotation.Nullable;
import javax.annotation.concurrent.GuardedBy;
import javax.annotation.concurrent.ThreadSafe;

import java.util.concurrent.Callable;
import java.util.function.Supplier;

import static com.google.common.base.Preconditions.checkState;
import static java.util.Objects.requireNonNull;

@ThreadSafe
public class SharedResource<T extends AutoCloseable>
{
    private final Callable<T> factory;
    @GuardedBy("this")
    private long counter;
    @GuardedBy("this")
    @Nullable
    private T instance;

    public SharedResource(Callable<T> factory)
    {
        this.factory = requireNonNull(factory, "factory is null");
    }

    public synchronized Lease<T> getInstanceLease()
            throws Exception
    {
        if (instance == null) {
            checkState(counter == 0, "Expected counter to be 0, but was: %s", counter);
            instance = requireNonNull(factory.call(), "factory.call() is null");
        }
        counter++;
        return new Lease<T>(instance, this::returnLease);
    }

    private synchronized Void returnLease()
            throws Exception
    {
        counter--;
        checkState(counter >= 0, "Counter got negative");
        if (counter == 0) {
            requireNonNull(instance, "instance is null")
                    .close();
            instance = null;
        }
        return null;
    }

    public static class Lease<T>
            implements AutoCloseable, Supplier<T>
    {
        private final T instance;
        private final Callable<Void> onClose;

        public Lease(T instance, Callable<Void> onClose)
        {
            this.instance = requireNonNull(instance, "instance is null");
            this.onClose = requireNonNull(onClose, "onClose is null");
        }

        @Override
        public T get()
        {
            return instance;
        }

        @Override
        public void close()
                throws Exception
        {
            onClose.call();
        }
    }
}
```

Reply:
Sure, I'll have a try on this